scipy/signal/tests/test_ltisys.py

```diff
@@ -1091,6 +1091,15 @@ def test_missing_BD_fails(self):
     def test_missing_CD_fails(self):
         assert_raises(ValueError, abcd_normalize, A=self.A, B=self.B)
 
+    def test_int_becomes_float(self):
+        # regression test for gh-13125
+        ABCD = [np.atleast_2d(m) for m in (1, 2, 3, 4)]
+        A, B, C, D = abcd_normalize(*ABCD)
+        assert_(np.issubdtype(A.dtype, np.inexact))
+        assert_(np.issubdtype(B.dtype, np.inexact))
+        assert_(np.issubdtype(C.dtype, np.inexact))
+        assert_(np.issubdtype(D.dtype, np.inexact))
+
```

Comment:
This is indeed still an issue on latest `main` branch. I wrote a test locally to more closely match the original reproducer:

<details>

```diff
--- a/scipy/signal/tests/test_ltisys.py
+++ b/scipy/signal/tests/test_ltisys.py
@@ -1,7 +1,7 @@
 import warnings
 
 import numpy as np
-from numpy.testing import suppress_warnings
+from numpy.testing import suppress_warnings, assert_allclose
 import pytest
 from pytest import raises as assert_raises
 from scipy._lib._array_api import(
@@ -1223,3 +1223,12 @@ class Test_freqresp:
         expected = 1 / (s + 1)**4
         assert_almost_equal(H.real, expected.real)
         assert_almost_equal(H.imag, expected.imag)
+
+
+def test_gh_13125():
+    sys = StateSpace(-4, 2, 1, 0)
+    sys2 = StateSpace(-4., 2, 1, 0)
+
+    _, y_expected = step(sys2)
+    _, y_actual = step(sys)
+    assert_allclose(y_actual, y_expected)
```

</details>

Without a domain expert review here I'm a bit concerned for the reason @roryyorke already brought up above re: type handling backwards compat.

In particular, on latest `main` with the changes here transplanted over I see a failure in `TestC2dLti.test_c2d_ss ` because it expects a type preservation/pass-through behavior.

I wonder who we could ping with about two weeks until we plan to branch SciPy `1.15.0`... From the `git blame` it looks like @befelix was the last person to do a lot of work in this area and they still seem somewhat-active on GitHub, but it has been several years.